### PR TITLE
Remove support for sorting terms aggregation by ascending count

### LINKED ISSUE #17614: Remove support for sorting terms aggregation by ascending count
We try to be as flexible as possible when it comes to sorting terms aggregations. However, sorting by anything but by _term or descending _count makes it very hard to return the correct top buckets and counts, which is disappointing to users. For this reason we should remove the ability to sort the terms aggregation by ascending count (split out from https://github.com/elastic/elasticsearch/issues/17588)

### PR DESCRIPTION
**Pull request by jimczi**
closes #17614

### CODE REVIEW

File: docs/reference/migration/migrate_5_0/aggregations.asciidoc [hunk 1]
```diff
@@ -26,3 +26,7 @@ for `from` and `to` anymore.
 `size: 0` is no longer valid for the terms, significant terms and geohash grid
 aggregations. Instead a size should be explicitly specified with a number greater
 than zero.
+
+==== `order: {"count": "desc"}` on Terms
```
Comment (jpountz) [inline docs/reference/migration/migrate_5_0/aggregations.asciidoc]: should it be s/desc/asc/ ?

**Comment by jpountz** [comment]
LGTM, can you put a comment on https://github.com/elastic/kibana/issues/6833 saying we are moving forward on this?

File: docs/reference/migration/migrate_5_0/aggregations.asciidoc (hunk 1 above)
Comment (jimferenczi) [inline docs/reference/migration/migrate_5_0/aggregations.asciidoc]: Sure, thanks !

**Comment by jimferenczi** [comment]
Sure, thanks @jpountz !

**Comment by jpountz** [comment]
@jimferenczi Let's add deprecation logging on 2.4/2.x?

**Comment by jimferenczi** [comment]
The functionality is deprecated in 2.4/2.x but this PR has been reverted since it requires changes in Kibana. I'll merge it after the alpha release.

**Pull request merged**